### PR TITLE
Merge branch RC1 to master

### PR DESCRIPTION
**Pull request by piponazo**
Creating this PR for @clanmills.

TODO: @clanmills feel free to edit the description to explain the scope of this PR.

### CODE REVIEW

**Comment by D4N** [comment]
It's going to take me a while to review everything here as there's a lot of changes (git reports `458 insertions(+), 1455 deletions(-)`).

Also, before we merge this, we should squash all commits prefixed with `WIP` and remove the `WIP` prefix.

**Comment by piponazo** [comment]
I'm going to dedicate some time to squish the commits now.

**Comment by piponazo** [comment]
I tried to squash the commits as much as possible in groups dealing with different topics. The number of commits went from 27 to 11. It would be difficult to squash even more the commits without doing mistakes in the process.

Most of the changes are about documentation and the buildserver scripts. Only few things are involved with the c++ code.

**Review by piponazo — COMMENTED** [review]
I did not go through the buildserver scripts, but I have some comments (mainly about the documentation)

**Review by clanmills — COMMENTED** [review]
This looks right to me.  For sure the *.md and *.sh file changes are correct and unique to RC1.

Can you look carefully at the Regex code.  I hammered something into RC1 earlier this week to get the stuff built and tested on MinGW, I think.  If in doubt, preserve 'master'.

I'd like to ask @D4N to approve this.

Once this is in 'master' and 'RC1' is in sync with 'master', I'll probably generate a daily PR for changes until we release Exiv2 v0.27 RC1 (which I hope will be about October 26).  How happy we shall be!

File: cmake/packaging.cmake [hunk 1]
```diff
@@ -2,7 +2,49 @@ set(CPACK_PACKAGE_NAME "${PROJECT_NAME}")
...
+elseif ( MSVC )
+    set(CPACK_GENERATOR ZIP)  # use .zip - less likely to damage bin/exiv2.dll permissions
+else()
+    set(CPACK_GENERATOR TGZ)  # Linux or MacOS-X: use .tar.gz (works with tar xzf bundle.tar.gz)
+endif()
+
+set(CPACK_PACKAGE_FILE_NAME ${CPACK_PACKAGE_NAME}-${CPACK_PACKAGE_VERSION}-${PACKNAME})
+
+# https://stackoverflow.com/questions/17495906/copying-files-and-including-them-in-a-cpack-archive
+install(DIRECTORY "${PROJECT_SOURCE_DIR}/samples/" DESTINATION "samples")
```
[the remark this answers is not in the document]
Comment (clanmills) [reply cmake/packaging.cmake]: In v0.26, the samples directory is used by 'dailyTest.sh' which compiles and links samples/exifprint.cpp and runs it.  It does that to be 100% certain that users can build and link code.  I think dailyTest.sh might test that MinGW/32 build links and runs with Qt.

I don't recall dailyTest.sh ever failing, so maybe should remove samples.

Let's leave it for now.  As I mentioned on Slack, Jenkins is now performing the builds.  I have to inspect and test the bundles.
```
596 rmills@rmillsmm:~/gnu/github/exiv2/exiv2 $ contrib/buildserver/build.sh --status all
-rw-rw-r-- 1 rmills rmills 2329605 Oct 16 20:41 exiv2-0.27.0.1-Linux.tar.gz
-rw-r--r-- 1 rmills  staff  3042158 16 Oct 20:42 exiv2-0.27.0.1-Darwin.tar.gz
-rw-r--r-- 1 rmills None 6152725 Oct 16 20:50 exiv2-0.27.0.1-MinGW.tar.gz
-rw-r--r-- 1 rmills None 3681422 Oct 16 18:21 exiv2-0.27.0.1-MinGW.tar.gz
-rw-r--r-- 1 rmills None 4130646 Oct 16 20:50 exiv2-0.27.0.1-CYGWIN.tar.gz
-rw-r--r-- 1 rmills None 2767030 Oct 16 20:47 exiv2-0.27.0.1-msvc.zip
597 rmills@rmillsmm:~/gnu/github/exiv2/exiv2 $ 
```

File: README.md [hunk 2]
```diff
@@ -2,181 +2,422 @@
...
-	md5sum       http://www.microbrew.org/tools/md5sha1sum/
+[TOC](#TOC)
+<name id="3-2"></a>
+### 3.2 Support
+For new bug reports and feature requests, please open an issue in Github.
 
-## Troubleshooting
+[TOC](#TOC)
+<name id="4"></a>
+## 4 Platform Notes
```
[the remark this answers is not in the document]
Comment (clanmills) [reply README.md]: I've found the platform notes very useful to get the build machines correctly configured.

We could put this on the web and provide the URL.  However the platforms notes change with different versions.  By including it in README.md, we are reminded to review those with every release.  And the notes apply to the release for that build.

File: README.md [hunk 3]
```diff
@@ -2,181 +2,422 @@
...
+```
+
+[TOC](#TOC)
+<name id="2-5"></a>
+### 2.5 Dependencies
+
+You can choose to use dependent libraries using your platform's package installer.
+
+We also support conan which is especially useful for users of Visual Studio.
+See [README-CONAN](http://exiv2.dyndns.org/0.27.0.1/README-CONAN.md) for more information
```
[the remark this answers is not in the document]
Comment (clanmills) [reply README.md]: The URL is a 'cut'n'paste' from the top of the file.  Every URL needs attention.

File: README.md [hunk 4]
```diff
@@ -2,181 +2,422 @@
...
-     @@@#       ,/    s@@#  @@@@     @@@b       Volker       @Tuan@
-    ]@@@Abhinav@@\   /@@@\    \@@@Q  @@@Q       %@@@#      /@@@@Mahesh@@#
-   /@@Raphael@@@@@\ /@@@@@\     C++  Metadata  Library    /@Sridhar@@@v0.26\
-</code></pre>
+    $ sudo make uninstall
 
-# Exiv2
+[TOC](#TOC)
+<name id="2-4"></a>
+### 2.4 Build options
```
[the remark this answers is not in the document]
Comment (clanmills) [reply README.md]: It's impossible to keep it in sync.  Sure we could have something in the build scripts to find and update this section - more stuff to do - more stuff to go wrong.  I'll review and simplify this.

About the "standard locations" (/usr/local etc).  You're right.  It shouldn't be there.  I think I got it from Andreas' good old ReadMe or something.  I'll remove that.

File: README.md [hunk 5]
```diff
@@ -2,181 +2,422 @@
...
+$ cmake -DBUILD_SHARED_LIBS=On -DEXIV2_ENABLE_NLS=OFF
+```
+
+[TOC](#TOC)
+<name id="2-5"></a>
+### 2.5 Dependencies
+
+You can choose to use dependent libraries using your platform's package installer.
+
+We also support conan which is especially useful for users of Visual Studio.
```
[the remark this answers is not in the document]
Comment (clanmills) [reply README.md]: Can we use conan on MinGW and Cygwin?  For, in README-CONAN.md, I've said not to use it.

**Pull request merged**